cmd/thanos/bucket.go

```diff
@@ -56,11 +75,18 @@ func registerBucket(m map[string]setupFunc, app *kingpin.Application, name strin
 	lsOutput := ls.Flag("ouput", "format in which to print each block's information; may be 'json' or custom template").
 		Short('o').Default("").String()
 
-	m[name+" ls"] = func(g *run.Group, logger log.Logger, _ *prometheus.Registry, _ opentracing.Tracer) error {
+	m[name+" ls"] = func(g *run.Group, logger log.Logger, reg *prometheus.Registry, _ opentracing.Tracer) error {
+		bkt, closeFn, err := getBucketClient(gcsBucket, s3config, reg)
+		if err != nil {
+			return err
+		}
+
 		// Dummy actor to immediately kill the group after the run function returns.
 		g.Add(func() error { return nil }, func(error) {})
 
-		return runBucketList(*gcsBucket, *lsOutput)
+		defer closeFn()
```

Comment:
this is `nil` for s3 client isn't? And will result in panic. Am I right?

Reply:
It gets defined by default as `closeFn = func() error { return nil }` on L254, should be alright.

Reply:
Oh men.. Returning all explicitly in each case on `getBucketClient` would be more readable. (: But not a blocker.